sentry-types/src/protocol/envelope.rs

```diff
@@ -1,16 +1,82 @@
-use std::io::Write;
+use std::{io::Write, path::Path};
 
+use serde::Deserialize;
+use thiserror::Error;
 use uuid::Uuid;
 
 use super::v7::{Attachment, Event, SessionAggregates, SessionUpdate, Transaction};
 
+/// Raised if a envelope cannot be parsed from a given input.
+#[derive(Debug, Error)]
+pub enum EnvelopeError {
+    /// Unexpected end of file
+    #[error("unexpected end of file")]
+    UnexpectedEof,
+    /// Missing envelope header
+    #[error("missing envelope header")]
+    MissingHeader,
+    /// Missing item header
+    #[error("missing item header")]
+    MissingItemHeader,
+    /// Missing newline after header or payload
+    #[error("missing newline after header or payload")]
+    MissingNewline,
+    /// Invalid envelope header
+    #[error("invalid envelope header")]
+    InvalidHeader(#[source] serde_json::Error),
+    /// Invalid item header
+    #[error("invalid item header")]
+    InvalidItemHeader(#[source] serde_json::Error),
+    /// Invalid item payload
+    #[error("invalid item payload")]
+    InvalidItemPayload(#[source] serde_json::Error),
+}
+
+#[derive(Deserialize)]
+struct EnvelopeHeader {
+    event_id: Option<Uuid>,
+}
+
+/// An Envelope Item Type.
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize)]
+enum EnvelopeItemType {
+    /// An Event Item type.
+    #[serde(rename = "event")]
+    Event,
+    /// A Session Item type.
+    #[serde(rename = "session")]
+    SessionUpdate,
+    /// A Session Aggregates Item type.
+    #[serde(rename = "sessions")]
+    SessionAggregates,
+    /// A Transaction Item type.
+    #[serde(rename = "transaction")]
+    Transaction,
+    /// An Attachment Item type.
+    #[serde(rename = "attachment")]
+    Attachment,
+}
+
+/// An Envelope Item Header.
+#[allow(dead_code)]
```

Comment:
```suggestion
```
We *are* using this, maybe we are not exposing the transitive users publicly?

Reply:
It was due to unused `attachment_type`.  Fixed it by using the value from item header and implementing `FromStr` for an `AttachmentType`